### ISSUE TITLE
[Bug] ImportError: libtorch_cuda_cu.so: cannot open shared object file: No such file or directory

### ISSUE THREAD

**Issue by gvalvano** [open-mmlab/mmagic#1854]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmagic/issues) and [Discussions](https://github.com/open-mmlab/mmagic/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmagic.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (main)](https://github.com/open-mmlab/mmagic) or [latest version (0.x)](https://github.com/open-mmlab/mmagic/tree/0.x).

### Task

I'm using the official example scripts/configs for the officially supported tasks/models/datasets.

### Branch

main branch https://github.com/open-mmlab/mmagic

### Environment

I'm not able to start the training of the model, and I cannot figure out why. 

When I launch:
```
CUDA_VISIBLE_DEVICES=1 python mmagic/tools/train.py ./custom_configs/dreambooth.py --work-dir work_dirs/example
```
I get the import error: "ImportError: libtorch_cuda_cu.so: cannot open shared object file: No such file or directory"

**My environment:**

> CUDA Version: 11.7 
>  
> Python 3.8.13
> 
> torch                             1.13.0a0+340c412
> torch-tensorrt                    1.1.0a0
> torchtext                         0.13.0a0
> torchvision                       0.13.0a0
> 
> mmagic                            1.0.1.dev0
> mmcv                              2.0.0
> mmcv-full                         1.7.1
> mmdet                             3.0.0
> mmengine                          0.7.3
>

I installed `mmagic` from source, as documented [here](https://mmagic.readthedocs.io/en/latest/get_started/install.html?highlight=install):
```
git clone https://github.com/open-mmlab/mmagic.git
cd mmagic
pip3 install -e . -v
```
I installed `mmcv` and  `mmengine` using _pip_. I also tried using _mim_, but without any difference.

### Reproduces the problem - code sample

-

### Reproduces the problem - command or script

CUDA_VISIBLE_DEVICES=1 python mmagic/tools/train.py ./configs/dreambooth/dreambooth.py --work-dir work_dirs/example

### Reproduces the problem - error message

Traceback (most recent call last):
  File "mmagic/tools/train.py", line 114, in <module>
    main()
  File "mmagic/tools/train.py", line 101, in main
    runner = Runner.from_cfg(cfg)
  File "/opt/conda/lib/python3.8/site-packages/mmengine/runner/runner.py", line 439, in from_cfg
    runner = cls(
  File "/opt/conda/lib/python3.8/site-packages/mmengine/runner/runner.py", line 406, in __init__
    self.model = self.build_model(model)
  File "/opt/conda/lib/python3.8/site-packages/mmengine/runner/runner.py", line 813, in build_model
    model = MODELS.build(model)
  File "/opt/conda/lib/python3.8/site-packages/mmengine/registry/registry.py", line 548, in build
    return self.build_func(cfg, *args, **kwargs, registry=self)
  File "/opt/conda/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 250, in build_model_from_cfg
    return build_from_cfg(cfg, registry, default_args)
  File "/opt/conda/lib/python3.8/site-packages/mmengine/registry/build_functions.py", line 98, in build_from_cfg
    obj_cls = registry.get(obj_type)
  File "/opt/conda/lib/python3.8/site-packages/mmengine/registry/registry.py", line 442, in get
    self.import_from_location()
  File "/opt/conda/lib/python3.8/site-packages/mmengine/registry/registry.py", line 376, in import_from_location
    import_module(loc)
  File "/opt/conda/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 843, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/[PATH_TO_WORK_DIR]/mmagic/mmagic/models/__init__.py", line 6, in <module>
    from .editors import *  # noqa: F401, F403
  File "/[PATH_TO_WORK_DIR]/mmagic/mmagic/models/editors/__init__.py", line 5, in <module>
    from .basicvsr_plusplus_net import BasicVSRPlusPlusNet
  File "/[PATH_TO_WORK_DIR]/mmagic/mmagic/models/editors/basicvsr_plusplus_net/__init__.py", line 2, in <module>
    from .basicvsr_plusplus_net import BasicVSRPlusPlusNet
  File "/[PATH_TO_WORK_DIR]/mmagic/mmagic/models/editors/basicvsr_plusplus_net/basicvsr_plusplus_net.py", line 5, in <module>
    from mmcv.ops import ModulatedDeformConv2d, modulated_deform_conv2d
  File "/opt/conda/lib/python3.8/site-packages/mmcv/ops/__init__.py", line 2, in <module>
    from .active_rotated_filter import active_rotated_filter
  File "/opt/conda/lib/python3.8/site-packages/mmcv/ops/active_rotated_filter.py", line 10, in <module>
    ext_module = ext_loader.load_ext(
  File "/opt/conda/lib/python3.8/site-packages/mmcv/utils/ext_loader.py", line 13, in load_ext
    ext = importlib.import_module('mmcv.' + name)
  File "/opt/conda/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
ImportError: libtorch_cuda_cu.so: cannot open shared object file: No such file or directory

**Comment by Z-Fran** [comment]
Looks like you install mmcv without cuda ops. Please uninstall mmcv and mmcv-full. And then install torch-gpu and `mim install mmcv`

**Comment by gvalvano** [comment]
Hi @Z-Fran :) thanks for your help. I tried your solution but, unfortunately, it does not work. Torch is able to see all the GPUs correctly. This is the result of a simple test I ran:

```python
>> import torch
>> torch.cuda.is_available()
True
>> torch.rand(3).to("cuda")
tensor([0.7668, 0.0177, 0.2108], device='cuda:0')
```

I tried to install `mmcv` using `mim`, but it did not make any difference :(

**Comment by Z-Fran** [comment]
please use `mim install mmcv==2.0.0`

**Comment by gvalvano** [comment]
Great! :) Installing `mmcv` from source worked. I only had to make sure I installed the right version of `opencv`, or it would fail the import. I found [this thread](https://stackoverflow.com/questions/72706073/attributeerror-partially-initialized-module-cv2-has-no-attribute-gapi-wip-gs) on StackOverflow that helped me pick the right version:

```bash
pip install opencv-python==4.5.5.64 
```
All good now!
